---
title: "DGE_2022_NAA"
author: "Noah Auger"
date: "10/25/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


Necessary Packages
```{r}
suppressMessages(library(DESeq2))
suppressMessages(library(gplots))
suppressMessages(library(RColorBrewer))
suppressMessages(library(EnhancedVolcano))
suppressMessages(library(tidyverse))
suppressMessages(library(ggplot2))
suppressMessages(library(dplyr))
suppressMessages(library(tibble))
suppressMessages(library(RColorBrewer))
suppressMessages(library(calibrate))
suppressMessages(library(dichromat))
suppressMessages(library(IDPmisc))
suppressMessages(library(limma))
suppressMessages(library(cowplot))
suppressMessages(library(ggrepel))
suppressMessages(library(pheatmap))
suppressMessages(library(GO.db))
suppressMessages(library(GSEABase))
suppressMessages(library(GOstats))
suppressMessages(library(GOplot))
suppressMessages(library(stargazer))
suppressMessages(library(readr))
suppressMessages(library(readxl))
```



* Loading the raw count data with the data points: 12h, 1d, 3, 7d, 14R, 14A, 14P, 21d.
```{r}
    count_data <- read.delim(file = "counts.txt", header = TRUE, row.names = 1)
```


* Reorder the columns from the raw count data so that they match the order of the metadata frame (samples):
```{r}
#For all timepoints
  count_data_all <- count_data [,c(31,32,33,1,2,3,4,20,21,22,23,24,25,15,16,17,18,19,10,11,5,6,7,8,9,12,13,14,26,27,28,29,30)]
    samples_all<-read.table("pdata_corset.txt", header = 2, sep = '\t')
    
```



* Generating the DESeq2 data set:
```{r}
#For all timepoints.
    dds_all <- DESeqDataSetFromMatrix(
      countData = count_data_all,
      colData = samples_all,
      design = ~condition)
```



* Prefiltering -- keeping only the rows with at least 10 reads:
```{r}
#For all timepoints.
  keep_all <- rowSums(counts(dds_all)) >= 30
    dds_all <- dds_all[keep_all, ]

```



* Normalizing the counts: estimating the size factor:
```{r}
#For all timepoints.
  dds_all <- estimateSizeFactors(dds_all)
    sizeFactors(dds_all)

```


### Quality assessment of the count data:



* Applying variance stabilizing transformation (VST) to perform a log2 transformation of the normalized count data:
```{r, eval=TRUE, echo=TRUE}
#For all timepoints.
  vsd_data_all <- vst(dds_all, blind = TRUE)

```

* Performing principal component analysis (PCA) to check for batch effects:
```{r}
#For all timepoints.
  plotPCA(vsd_data_all, intgroup = "condition")

```









### Differential expression analysis


* Fitting and testing the model:
```{r}
dds_all <- DESeq(dds_all)
```


* Plotting the dispersion estimates:
```{r}
plotDispEsts(dds_all)
```










* Extract results of the differential gene expression analysis. Contrasts between conditions:
```{r}
res_12hpe <- results(dds_all, contrast = c("condition", "HPE_12", "Mesentery"))

res_1dpe <- results(dds_all, contrast = c("condition", "DPE_1", "Mesentery"))

res_3dpe <- results(dds_all, contrast = c("condition", "DPE_3", "Mesentery"))

res_7dpe <- results(dds_all, contrast = c("condition", "DPE_7", "Mesentery"))

res_14dpeR <- results(dds_all, contrast = c("condition", "DPE_14_R", "Mesentery"))

res_14dpeA <- results(dds_all, contrast = c("condition", "DPE_14_A", "Intestine"))

res_14dpeP <- results(dds_all, contrast = c("condition", "DPE_14_P", "Intestine"))

res_21dpe <- results(dds_all, contrast = c("condition", "DPE_21", "Intestine"))
```




Take the differential gene expression analysis results and turn it into a data frame:
```{r}
res_12hpe <- res_12hpe %>%
    as.data.frame() %>%
    rownames_to_column(var = "Clusters")


res_1dpe <- res_1dpe %>%
    as.data.frame() %>%
    rownames_to_column(var = "Clusters")

  
res_3dpe <- res_3dpe %>%
    as.data.frame() %>%
    rownames_to_column(var = "Clusters")


res_7dpe <- res_7dpe %>%
    as.data.frame() %>%
    rownames_to_column(var = "Clusters")


res_14dpeR <- res_14dpeR %>%
    as.data.frame() %>%
    rownames_to_column(var = "Clusters")


res_14dpeA <- res_14dpeA %>%
    as.data.frame() %>%
    rownames_to_column(var = "Clusters") 


res_14dpeP <- res_14dpeP %>%
    as.data.frame() %>%
    rownames_to_column(var = "Clusters")


res_21dpe <- res_21dpe %>%
    as.data.frame() %>%
    rownames_to_column(var = "Clusters")
```




Export the differentially expressed genes into a csv file:
```{r, eval=FALSE, echo = TRUE}
write.csv(res_12hpe, file = "DE_genes.csv", quote = FALSE, row.names = FALSE)
```


Extract the cluster IDs:
```{bash, eval = FALSE, echo = TRUE}
cat DE_genes.csv | cut -f1 -d',' | sed 1d > DE_cluster_IDs.txt
```


Extract the sequences corresponding to the clusters using the [fetchClusterSeqs.py](https://github.com/Adamtaranto/Corset-tools) utility:
```{bash, eval = FALSE, echo = TRUE}
../../../Corset-tools/fetchClusterSeqs.py -l -i /work/jgarcia/nauger/Transcriptome_Global_2022/cuke2022.trinity.Trinity.fasta -t DE_genes.txt -c clusters.txt -o interest.fasta
```


Run BlastX against the uniprot database:
```{bash, eval = FALSE, echo = TRUE}
blastx -query interest.fasta -db uniprot -out uniprot.blast -evalue 1e-5 -num_threads 16 -max_target_seqs 1 -outfmt 6 
```


Read blast annotation into R:
```{r}
blast <- read_tsv(file = "uniprot.blast", col_names = FALSE)
head(blast)
```


Get rid of unnecessary columns by leaving only query ID, blast hit ID, and the e-value:
```{r}
blast <- dplyr::select(blast, 1, 2, 11)
head(blast)
```


Separating the first column into two: Cluster names and Trinity names:
```{r}
blast <- separate(blast, col = 1, sep = "_", into = c("Cluster", "Trinity"), extra = "merge")
head(blast)
```


Modifying the third column now, keeping the UniprotID of the blast hit only, renaming the last column:
```{r}
blast <- blast %>%
  separate(col = 3, sep = "\\|", into = c("x", "BlastHitUniprotID", "y")) %>%
  dplyr::select(- c("x", "y"))
blast <- rename(blast, "BlastHitEvalue" = X11)
head(blast)
```


* Reduce redundancy by only keeping the the lowest evalue of a cluster. If a cluster has duplicates with the same Evalue, then distinct will only keep one.
```{r}
blast <- blast %>%
  group_by(Cluster) %>%
  filter(BlastHitEvalue == min(BlastHitEvalue)) %>%
  distinct(Cluster, .keep_all = T)
```


Extract Uniprot IDs, then upload IDs into ID Mapping function at Uniprot.org. This will give us gene names and GO terms, among other info.
```{r}
blast_UNIPRTO <- dplyr::select(blast, 3)
write.csv(blast_UNIPRTO,("blast_UNIPRTO.csv"))
```


Label Clusters with the gene names from the previous step.
```{r}
Uniprot_Annotations<-read_excel("uniprot-download_true_fields_accession_2Creviewed_2Cid_2Cprotein_nam-2022.11.02-19.28.44.60.xlsx")
head(Uniprot_Annotations)

Uniprot_Annotations <- blast %>% left_join(Uniprot_Annotations, by = c("BlastHitUniprotID"="From"))
head(Uniprot_Annotations)
```


Load in the name map and annotations from S. purpuratus that was run on Dammit:
```{r}
GeneNames<-read.csv("Genes_infotags.csv")
Final_NameMap <- read.csv("Final_NameMap.csv")
```


Reduce redundancy by only keeping the the lowest evalue of a cluster. If a cluster has duplicates with the same Evalue, then distinct will only keep one.
```{r}
GeneNames <- GeneNames %>%
  group_by(q_name) %>%
  filter(E == min(E)) %>%
  distinct(q_name, .keep_all = T)
```

```{r}
Purpuratus_Annotations <- Final_NameMap %>% left_join(GeneNames, by = c("Transcript"="q_name"), extra = "merge")
head(Purpuratus_Annotations)

Purpuratus_Annotations <- Purpuratus_Annotations %>% drop_na(E)

Purpuratus_Annotations <- separate(Purpuratus_Annotations, col = 15, sep = " ", into = c("Code", "Purpuratus Homolg"), extra = "merge")
head(Purpuratus_Annotations)
```


```{r}
HPE_12 <- Uniprot_Annotations %>% left_join(res_12hpe, by = c("Cluster"="Clusters"))
HPE_12 <- HPE_12 %>% left_join(Purpuratus_Annotations, by = c("Trinity"="Trinity"))
HPE_12 <- HPE_12 [,c(9,42,27,23,1,2,28,8,21,3,4,17,18,19,16,20,14)]
write.csv(HPE_12,("12 HPE.csv"))

DPE_1 <- Uniprot_Annotations %>% left_join(res_1dpe, by = c("Cluster"="Clusters"))
DPE_1 <- DPE_1 %>% left_join(Purpuratus_Annotations, by = c("Trinity"="Trinity"))
DPE_1 <- DPE_1 [,c(9,42,27,23,1,2,28,8,21,3,4,17,18,19,16,20,14)]
write.csv(DPE_1,("1 DPE.csv"))

DPE_3 <- Uniprot_Annotations %>% left_join(res_3dpe, by = c("Cluster"="Clusters"))
DPE_3 <- DPE_3 %>% left_join(Purpuratus_Annotations, by = c("Trinity"="Trinity"))
DPE_3 <- DPE_3 [,c(9,42,27,23,1,2,28,8,21,3,4,17,18,19,16,20,14)]
write.csv(DPE_3,("3 DPE.csv"))

DPE_7 <- Uniprot_Annotations %>% left_join(res_7dpe, by = c("Cluster"="Clusters"))
DPE_7 <- DPE_7 %>% left_join(Purpuratus_Annotations, by = c("Trinity"="Trinity"))
DPE_7 <- DPE_7 [,c(9,42,27,23,1,2,28,8,21,3,4,17,18,19,16,20,14)]
write.csv(DPE_7,("7 DPE.csv"))

DPE_14_R <- Uniprot_Annotations %>% left_join(res_14dpeR, by = c("Cluster"="Clusters"))
DPE_14_R <- DPE_14_R %>% left_join(Purpuratus_Annotations, by = c("Trinity"="Trinity"))
DPE_14_R <- DPE_14_R [,c(9,42,27,23,1,2,28,8,21,3,4,17,18,19,16,20,14)]
write.csv(DPE_14_R,("14 DPE Rudiment.csv"))

DPE_14_A <- Uniprot_Annotations %>% left_join(res_14dpeA, by = c("Cluster"="Clusters"))
DPE_14_A <- DPE_14_A %>% left_join(Purpuratus_Annotations, by = c("Trinity"="Trinity"))
DPE_14_A <- DPE_14_A [,c(9,42,27,23,1,2,28,8,21,3,4,17,18,19,16,20,14)]
write.csv(DPE_14_A,("14 DPE Anterior.csv"))

DPE_14_P <- Uniprot_Annotations %>% left_join(res_14dpeP, by = c("Cluster"="Clusters"))
DPE_14_P <- DPE_14_P %>% left_join(Purpuratus_Annotations, by = c("Trinity"="Trinity"))
DPE_14_P <- DPE_14_P [,c(9,42,27,23,1,2,28,8,21,3,4,17,18,19,16,20,14)]
write.csv(DPE_14_P,("14 DPE Posterior.csv"))

DPE_21 <- Uniprot_Annotations %>% left_join(res_21dpe, by = c("Cluster"="Clusters"))
DPE_21 <- DPE_21 %>% left_join(Purpuratus_Annotations, by = c("Trinity"="Trinity"))
DPE_21 <- DPE_21 [,c(9,42,27,23,1,2,28,8,21,3,4,17,18,19,16,20,14)]
write.csv(DPE_21,("21 DPE.csv"))
```



* Volcano Plot of Differentially expressed data at a certain time point. (Must use res_12hpe,1dpe, or 3dpe)
```{r}
colors <- rep("black", nrow(HPE_12))
names(colors) <- rep("NC", nrow(HPE_12))

colors[which(HPE_12$log2FoldChange >= 0.58 & HPE_12$padj < 0.05)] <- "red2"
names(colors) [which(HPE_12$log2FoldChange >= 0.58 & HPE_12$padj < 0.05)] <- "UP"

colors[which(HPE_12$log2FoldChange <= -0.58 & HPE_12$padj < 0.05)] <- "royalblue"
names(colors) [which(HPE_12$log2FoldChange <= 0.58 & HPE_12$padj < 0.05)] <- "DOWN"

EnhancedVolcano(
  HPE_12,
  lab = rownames(HPE_12),
  x = "log2FoldChange",
  y = "padj",
  pCutoff = 0.05,
  FCcutoff = 0.58,
  selectLab = "",
  colCustom = colors,
  colAlpha = 0.3,
  title = "",
  subtitle = "",
  caption = "",
  ylim = c(0, 20),
)
```


```{r}
Target_Genes<-read.csv("Target_Genes.csv", header = TRUE, row.names = 1)

Target_Genes <-Target_Genes$Clusters

Target_heatmap<-assay(vsd_data_all[c(print(Target_Genes))])


#Adding row annotations
write.csv(Target_heatmap,("Targets.csv"))

Target_heatmap<-read.csv("Targets.csv", header = TRUE, row.names = 1)
Target_heatmap<-log(Target_heatmap)
Target_heatmap<-as.matrix(Target_heatmap)

pathway_df = data.frame("." = Target_Genes$Wnt.Pathway)
rownames(pathway_df)= rownames(Target_heatmap)



#Print HeatMap
pheatmap(Target_heatmap, cluster_cols = F, cluster_rows = T, annotation_col = tissue_type_df, treeheight_row = 30, legend = F, annotation_legend = F, labels_col = "", cutree_rows = 4, annotation_row = pathway_df)

pheatmap(Target_heatmap, cluster_cols = F, cluster_rows = T, annotation_col = tissue_type_df, treeheight_row = 30, legend = F, annotation_legend = T, labels_col = "", cutree_rows = 2, annotation_row = pathway_df)
```



```{r}
#Extract the rows from the Gene Names column that have Wnt.
Wnt <- HPE_12[grep("Wnt", HPE_12$`Gene Names`, ignore.case = T), ]


#Extract the Cluster column from the Wnt data frame, then use them to get the count matrix from the vsd_data_all.
Wnt_Clusters<-Wnt$Cluster
Wnt_heatmap<-assay(vsd_data_all[c(print(Wnt_Clusters))])

#Download the Wnt_Heatmap matrix so that you can can add names to the cluster Ids and delete the rows that are not needed.
write.csv(Wnt_heatmap,("Wnt_heatmap.csv"))


#Reload the Wnt_heatmap so that you can scale the count values 
Wnt_heatmap<-read.csv("Wnt_heatmap.csv", header = TRUE, row.names = 1)
Wnt_heatmap<-log(Wnt_heatmap)
Wnt_heatmap<-as.matrix(Wnt_heatmap)


#Addiing Column annotations
tissue_type_df = data.frame("Tissue" = c(rep("Normal Mesentery (ct + me)",3),rep("Rudiment (ct + me)",17),rep("Rudiment + Lumen (ct + me + le)",8),rep("Normal Intestine (ct + me + le)",5)))
rownames(tissue_type_df) = colnames(Wnt_heatmap)


#Print HeatMap
pheatmap(Wnt_heatmap, cluster_cols = F, cluster_rows = T, annotation_col = tissue_type_df , cellheight = 17 , cellwidth = 12, treeheight_row = 30, legend = F, annotation_legend = F, labels_col = "", cutree_rows = 4)

pheatmap(Wnt_heatmap, cluster_cols = F, cluster_rows = F, annotation_col = tissue_type_df , cellheight = 17 , cellwidth = 12, treeheight_row = 30, legend = F, annotation_legend = F, show_colnames = F, cutree_rows = 2, gaps_col = c(3,7,10,13,18,20,22,25,28))

pheatmap(Wnt_heatmap, cluster_cols = F, cluster_rows = F, annotation_col = tissue_type_df, legend = T, annotation_legend = F, show_colnames = F, cutree_rows = 2, gaps_col = c(3,7,10,13,18,20,22,25,28))
```





```{r}
#Extract the rows from the Gene Names column that have Wnt.
DVl <- HPE_12[grep("Cluster-129860.0", HPE_12$`Cluster`, ignore.case = T), ]


#Extract the Cluster column from the Wnt data frame, then use them to get the count matrix from the vsd_data_all.
Fz_DVL<-DVl$Cluster
Fz_DVL<-assay(vsd_data_all[c(print(Fz_DVL))])


#Download the Wnt_Heatmap matrix so that you can can add names to the cluster Ids and delete the rows that are not needed.
write.csv(Fz_DVL,("Fz_DVL.csv"))


#Reload the Wnt_heatmap so that you can scale the count values 
Fz_heatmap<-read.csv("Fz_heatmap.csv", header = TRUE, row.names = 1)
Fz_heatmap<-log(Fz_heatmap)
Fz_heatmap<-as.matrix(Fz_heatmap)


#Heat ap
pheatmap(Fz_heatmap, cluster_cols = F, cluster_rows = F, annotation_col = tissue_type_df , cellheight = 17 , cellwidth = 12, treeheight_row = 30, legend = F, annotation_legend = F, show_colnames = F, gaps_col = c(3,7,10,13,18,20,22,25,28))

pheatmap(Fz_heatmap, cluster_cols = F, cluster_rows = F, annotation_col = tissue_type_df, legend = F, annotation_legend = F, show_colnames = F, gaps_col = c(3,7,10,13,18,20,22,25,28))

pheatmap(Fz_heatmap, cluster_cols = F, cluster_rows = T, annotation_col = tissue_type_df, labels_col = "")

pheatmap(Wnt_heatmap, cluster_cols = F, cluster_rows = F, annotation_col = tissue_type_df, legend = T, annotation_legend = F, show_colnames = F, cutree_rows = 2, gaps_col = c(3,7,10,13,18,20,22,25,28))
```





Fz and Wnt
```{r}
Wnt_Fz_heatmap<-read.csv("Wnt_Fz_heatmap.csv", header = TRUE, row.names = 1)
Wnt_Fz_heatmap<-log(Wnt_Fz_heatmap)
Wnt_Fz_heatmap<-as.matrix(Wnt_Fz_heatmap)

#Different color palettes: color = colorRampPalette(c("blue4", "white", "red4"))(100)) & color=colorRampPalette(c("navy","white","red"))(50))

#Heat ap

pheatmap(Wnt_Fz_heatmap, cluster_cols = F, cluster_rows = T, annotation_col = tissue_type_df, treeheight_row = 10, labels_col = "")

pheatmap(Wnt_Fz_heatmap, cluster_cols = F, cluster_rows = F, annotation_col = tissue_type_df , cellheight = 17 , cellwidth = 12, treeheight_row = 30, legend = F, annotation_legend = F, show_colnames = F, gaps_col = c(3,7,10,13,18,20,22,25,28))

pheatmap(Wnt_Fz_heatmap, cluster_cols = F, cluster_rows = T, annotation_col = tissue_type_df, treeheight_row = 30, legend = T, annotation_legend = T, labels_col = "", show_colnames = F, gaps_col = c(3,7,10,13,18,20,22,25,28))
```



All
```{r}
Wnt_Fz_Targets_heatmap.csv

Wnt_Fz_Targets_heatmap<-read.csv("Wnt_Fz_Targets_heatmap.csv", header = TRUE, row.names = 1)
Wnt_Fz_Targets_heatmap<-log(Wnt_Fz_Targets_heatmap)
Wnt_Fz_Targets_heatmap<-as.matrix(Wnt_Fz_Targets_heatmap)


#Print HeatMap
pheatmap(Wnt_Fz_Targets_heatmap, cluster_cols = F, cluster_rows = F, annotation_col = tissue_type_df, treeheight_row = 30, legend = F, annotation_legend = T, labels_col = "", cutree_rows = 2, annotation_row = pathway_df)

pheatmap(Wnt_Fz_Targets_heatmap, cluster_cols = F, cluster_rows = T, annotation_col = tissue_type_df, treeheight_row = 30, legend = F, annotation_legend = F, annotation_row = pathway_df, show_colnames = F, fontsize= 8, gaps_col = c(3,7,10,13,18,20,22,25,28), cutree_rows = 4)

pheatmap(Wnt_Fz_Targets_heatmap, cluster_cols = F, cluster_rows = T, annotation_col = tissue_type_df, treeheight_row = 30, legend = F, annotation_legend = F, annotation_row = pathway_df, labels_col = "", cutree_rows = 4, fontsize= 8)

pheatmap(Wnt_Fz_Targets_heatmap, cluster_cols = F, cluster_rows = T, annotation_col = tissue_type_df, treeheight_row = 30, legend = F, annotation_legend = F, annotation_row = pathway_df, cutree_rows = 4)

pheatmap(Wnt_Fz_Targets_heatmap, cluster_cols = F, cluster_rows = T, annotation_col = tissue_type_df, treeheight_row = 10, labels_col = "")
```



Wnt signaling pathway components
```{r}
Beta<-read.csv("B-catenin Pathway.csv", header = TRUE, row.names = 1)

Beta_clusters <-Beta$Cluster

Beta_heatmap<-assay(vsd_data_all[c(print(Beta_clusters))])


#Adding row annotations
write.csv(Beta_heatmap,("Beta_heatmap.csv"))


Beta_heatmap<-read.csv("Beta_heatmap.csv", header = TRUE, row.names = 1)
Beta_heatmap<-log(Beta_heatmap)
Beta_heatmap<-as.matrix(Beta_heatmap)

Wnt_pathway_df = data.frame("." = Beta$Component)
rownames(Wnt_pathway_df)= rownames(Beta_heatmap)



#Print HeatMap
pheatmap(Beta_heatmap, cluster_cols = F, cluster_rows = T, annotation_col = tissue_type_df, treeheight_row = 30, legend = F, annotation_legend = F, cutree_rows = 4, annotation_row = Wnt_pathway_df)

pheatmap(Beta_heatmap, cluster_cols = F, cluster_rows = T, annotation_col = tissue_type_df, treeheight_row = 30, legend = T, annotation_legend = T, labels_col = "", cutree_rows = 2, annotation_row = Wnt_pathway_df)

pheatmap(Beta_heatmap, cluster_cols = F, cluster_rows = T, annotation_col = tissue_type_df, treeheight_row = 30, legend = T, annotation_legend = T, annotation_row = Wnt_pathway_df,labels_col = "", show_colnames = F, fontsize= 8, gaps_col = c(3,7,10,13,18,20,22,25,28), annotation_colors = list(Component = c("Off-state"="blue", "On-state"="cyan", "Target_Gene"="red")))

pheatmap(Beta_heatmap, cluster_cols = F, cluster_rows = F, annotation_col = tissue_type_df, treeheight_row = 30, legend = F, annotation_legend = F, annotation_row = Wnt_pathway_df,labels_col = "", show_colnames = F, fontsize= 8, gaps_col = c(3,7,10,13,18,20,22,25,28), annotation_colors = list(Component = c("Off-state"="blue", "On-state"="cyan", "Target_Gene"="red")))

pheatmap(Beta_heatmap, cluster_cols = F, cluster_rows = T, annotation_col = tissue_type_df, treeheight_row = 30, legend = F, annotation_legend = F, annotation_row = Wnt_pathway_df,labels_col = "", show_colnames = F, gaps_col = c(3,7,10,13,18,20,22,25,28), annotation_colors = list(Component = c("Off-state"="blue", "On-state"="cyan", "Target_Gene"="red")))
```



PCP pathway
```{r}
PCP<-read.csv("PCP.csv", header = TRUE, row.names = 1)

PCP_clusters <-PCP$Cluster

PCP_heatmap<-assay(vsd_data_all[c(print(PCP_clusters))])


#Adding row annotations
write.csv(PCP_heatmap,("PCP_heatmap.csv"))


PCP_heatmap<-read.csv("PCP_heatmap.csv", header = TRUE, row.names = 1)
PCP_heatmap<-log(PCP_heatmap)
PCP_heatmap<-as.matrix(PCP_heatmap)

PCP_pathway_df = data.frame("." = PCP$Component)
rownames(PCP_pathway_df)= rownames(PCP_heatmap)


#Print HeatMap
pheatmap(PCP_heatmap, cluster_cols = F, cluster_rows = T, annotation_col = tissue_type_df, treeheight_row = 30, legend = F, annotation_legend = F, cutree_rows = 4, annotation_row = PCP_pathway_df)

pheatmap(PCP_heatmap, cluster_cols = F, cluster_rows = T, annotation_col = tissue_type_df, treeheight_row = 30, legend = T, annotation_legend = F, labels_col = "", cutree_rows = 2, annotation_row = PCP_pathway_df)

pheatmap(PCP_heatmap, cluster_cols = F, cluster_rows = T, annotation_col = tissue_type_df, treeheight_row = 30, legend = T, annotation_legend = T, annotation_row = PCP_pathway_df,labels_col = "", show_colnames = F, fontsize= 8, gaps_col = c(3,7,10,13,18,20,22,25,28))

pheatmap(PCP_heatmap, cluster_cols = F, cluster_rows = T, annotation_col = tissue_type_df, treeheight_row = 30, legend = F, annotation_legend = F, annotation_row = PCP_pathway_df,labels_col = "", show_colnames = F, fontsize= 8, gaps_col = c(3,7,10,13,18,20,22,25,28))

pheatmap(PCP_heatmap, cluster_cols = F, cluster_rows = F, annotation_col = tissue_type_df, legend = F, annotation_legend = F, annotation_row = PCP_pathway_df,labels_col = "", show_colnames = F, gaps_col = c(3,7,10,13,18,20,22,25,28))
```
